File: backend/app/schemas/weather.py
from typing import Literal

from pydantic import BaseModel


class WeatherData(BaseModel):
    temp: float
    condition: Literal["sunny", "rainy", "cloudy", "storm", "drought"]
    humidity: int
    windSpeed: float
    icon: str
    description: str


class Alert(BaseModel):
    id: str
    type: Literal["weather", "pest", "drought", "flood"]
    severity: Literal["low", "medium", "high", "critical"]
    title: str
    description: str
    timestamp: str
    icon: str

File: src/api/weather.ts
import { apiFetch } from './client';

export interface WeatherData {
  temp: number;
  condition: 'sunny' | 'rainy' | 'cloudy' | 'storm' | 'drought';
  humidity: number;
  windSpeed: number;
  icon: string;
  description: string;
}

export interface Alert {
  id: string;
  type: 'weather' | 'pest' | 'drought' | 'flood';
  severity: 'low' | 'medium' | 'high' | 'critical';
  title: string;
  description: string;
  timestamp: string;
  icon: string;
}

export async function getWeather(): Promise<WeatherData> {
  return apiFetch<WeatherData>('/weather');
}

export async function getAlerts(): Promise<Alert[]> {
  return apiFetch<Alert[]>('/alerts');
}
